### PR TITLE
CB-13725: Datahub Autoscaling History endpoint should support downloading events as zip

### PR DESCRIPTION
**Pull request by juhi-09**

### CODE REVIEW

**Comment by horadla23** [comment]
for me it seems this PR is draft, commit message should be fixed, also CB won't start using this implementation, so please change this PR to draft, until it is tested

File: core/build.gradle [hunk 1]
```diff
@@ -178,6 +178,8 @@ dependencyManagement {
     dependency group: 'org.apache.commons',                 name: 'commons-collections4',        version: commonsCollections4Version
     dependency group: 'org.apache.commons',                 name: 'commons-lang3',               version: apacheCommonsLangVersion
     dependency group: 'com.cloudera.cdp',                   name: 'cdp-sdk-java',                version: cdpSdkVersion
+    dependency group: 'org.apache.hadoop',                  name: 'hadoop-yarn-api',           version: hadoopVersion
```
Comment (juhi-09) [inline core/build.gradle:181]: `hadoop-yarn-api` and `consul-api` are added as dependencies of `autoscale` package.

File: core/src/main/resources/messages/messages.properties [hunk 2]
```diff
@@ -0,0 +1,472 @@
+test.message=Hi my dear friend
```
Comment (juhi-09) [inline core/src/main/resources/messages/messages.properties:1]: The message.properties are added for core package to access the messages for `CloudbreakMessagesHostServiceTypeTest`

Comment (lajosrodek) [reply core/src/main/resources/messages/messages.properties:1]: Sorry, I really do not get this one. The same messages are already available in [the similarly named](https://github.com/hortonworks/cloudbreak/blob/master/common/src/main/resources/messages/messages.properties) file of the `common` module.

File: core/build.gradle [hunk 3]
```diff
@@ -287,6 +290,7 @@ dependencies {
   implementation project(':auth-connector')
   implementation project(':secret-engine')
   implementation project(':template-manager-core')
+  implementation project(':autoscale')
```
Comment (lajosrodek) [inline core/build.gradle:293]: No, it is a very bad idea to add dependencies between microservice modules. You shall rely on `autoscale-api` that contains the respective client and endpoint types.

File: core/src/main/java/com/sequenceiq/cloudbreak/controller/v4/AuditEventV4Controller.java [hunk 4]
```diff
@@ -42,18 +49,25 @@ public AuditEventV4Responses getAuditEvents(Long workspaceId, String resourceTyp
 
     }
 
+    public List<AutoscaleClusterHistoryResponse> getAutoscalingEvents(String resourceCrn) {
+        return historyController.getHistoryByCrn(resourceCrn, MAX_AUTOSCALING_EVENTS);
```
Comment (lajosrodek) [inline core/src/main/java/com/sequenceiq/cloudbreak/controller/v4/AuditEventV4Controller.java:53]: This call will not work. You shall instead rely on the respective autoscale client and endpoint. Remember, you are about to make a REST call between two microservices!

**Review by lajosrodek — CHANGES_REQUESTED** [review]
A few initial notes.

**Comment by doktoric** [comment]
@lajosrodek @juhi-09  do we still need this ?

**Comment by lajosrodek** [comment]
@doktoric I do not have a clue to be honest. :smile:

**Comment by lnardai** [comment]
Looks abandoned. Please reopen if needed.

**Pull request closed without merging**